DOC: ongoing fixing of doctests

Starting with https://github.com/pandas-dev/pandas/pull/19952, we run (some) doctests on Travis to ensure the example are kept up to date. 
Currently, we only test the docstrings for Series and DataFrame, and reshape-related top level functions, *and* we also skip a lot of the tests.

So PRs to incrementally test more of the docstrings are very welcome, and can be good bite-sized newbie issues.

The workflow would be like this:
- Take a certain function or a certain module

- Fix the examples in the docstring(s):
  - To run the doctest for a single method, you can either use the validation script, eg:

    ```
    python scripts/validate_docstrings.py pandas.DataFrame.mean
    ```

    to run a full module or a subset of them, you can use something like:

    ```
    pytest --doctest-modules pandas/core/frame.py -k DataFrame
    ```

  - It is certainly welcome to also fix other problems with the docstring (all other errors and warnings that the validation script gives), or to actually improve the example. But I think it is also fine to do PRs just fixing the doctests as they are, so we at least can test them (so if that then somebody else improves the examples, we are sure those new examples are actually tested).
  
  - Some more details can be found here: http://pandas-docs.github.io/pandas-docs-travis/contributing.html#updating-a-pandas-docstring. And specifically, some tips on getting doctests to pass: http://pandas-docs.github.io/pandas-docs-travis/contributing_docstring.html#tips-for-getting-your-examples-pass-the-doctests
  
- When the doctests pass for a certain function or module, add those (or remove the skip) to the ones that are tested on Travis. 
  For that, you need to edit the `ci/code_checks.sh` file. For example, for the Series tests, it looks like:
  
  https://github.com/pandas-dev/pandas/blob/cf11f71e423be65e9c2ae83ea1bab8b69ee79ca4/ci/code_checks.sh#L125-L126
  
  Here, you could remove one of the skipped ones (the `-k"-agg"` syntax is to skip the doctests for Series.agg and Series.aggregate). Or add such a new line for a new module.
  
  

Good docstrings to start with are the ones for Series and DataFrame (the ones located in `pandas/core/series.py`, `pandas/core/frame.py` and `pandas/core/generic.py`). 
After that, adding new modules like the indexing ones would be a possibility.

> i see a lot of merged PRs, is this done?

Looks like there are still some docstrings with errors:
```
FAILED pandas/_config/config.py::pandas._config.config.option_context
FAILED pandas/_testing/__init__.py::pandas._testing._make_timeseries
FAILED pandas/_testing/_io.py::pandas._testing._io.network
FAILED pandas/_testing/asserters.py::pandas._testing.asserters._get_tol_from_less_precise
FAILED pandas/_testing/contexts.py::pandas._testing.contexts.set_timezone
FAILED pandas/core/flags.py::pandas.core.flags.Flags.allows_duplicate_labels
FAILED pandas/core/resample.py::pandas.core.resample.Resampler.aggregate
FAILED pandas/core/resample.py::pandas.core.resample.Resampler.transform
FAILED pandas/core/arrays/datetimelike.py::pandas.core.arrays.datetimelike.DatelikeOps.strftime
FAILED pandas/core/internals/construction.py::pandas.core.internals.construction.dataclasses_to_dicts
FAILED pandas/io/stata.py::pandas.io.stata.read_stata
FAILED pandas/io/formats/format.py::pandas.io.formats.format.EngFormatter.__call__
FAILED pandas/io/formats/style.py::pandas.io.formats.style.Styler.apply
FAILED pandas/io/formats/style.py::pandas.io.formats.style.Styler.applymap
FAILED pandas/io/formats/style.py::pandas.io.formats.style.Styler.background_gradient
FAILED pandas/io/formats/style.py::pandas.io.formats.style.Styler.hide_columns
FAILED pandas/io/formats/style.py::pandas.io.formats.style.Styler.hide_index
FAILED pandas/io/formats/style.py::pandas.io.formats.style.Styler.highlight_between
FAILED pandas/io/formats/style.py::pandas.io.formats.style.Styler.highlight_quantile
FAILED pandas/io/formats/style.py::pandas.io.formats.style.Styler.pipe
FAILED pandas/io/formats/style.py::pandas.io.formats.style.Styler.set_properties
FAILED pandas/io/formats/style.py::pandas.io.formats.style.Styler.set_table_attributes
FAILED pandas/io/formats/style.py::pandas.io.formats.style.Styler.set_table_styles
FAILED pandas/io/formats/style.py::pandas.io.formats.style.Styler.set_td_classes
FAILED pandas/io/formats/style.py::pandas.io.formats.style.Styler.set_tooltips
FAILED pandas/io/formats/style.py::pandas.io.formats.style.Styler.text_gradient
FAILED pandas/io/formats/style.py::pandas.io.formats.style.Styler.to_latex
FAILED pandas/io/formats/style.py::pandas.io.formats.style.Styler.where
FAILED pandas/io/formats/style_render.py::pandas.io.formats.style_render.StylerRenderer.format
FAILED pandas/io/formats/style_render.py::pandas.io.formats.style_render._parse_latex_header_span
FAILED pandas/io/formats/style_render.py::pandas.io.formats.style_render._parse_latex_table_styles
FAILED pandas/plotting/_core.py::pandas.plotting._core.PlotAccessor.line
FAILED pandas/plotting/_core.py::pandas.plotting._core.boxplot
FAILED pandas/plotting/_core.py::pandas.plotting._core.boxplot_frame
FAILED pandas/plotting/_core.py::pandas.plotting._core.boxplot_frame_groupby
FAILED pandas/plotting/_misc.py::pandas.plotting._misc.andrews_curves
FAILED pandas/plotting/_misc.py::pandas.plotting._misc.autocorrelation_plot
FAILED pandas/plotting/_misc.py::pandas.plotting._misc.bootstrap_plot
FAILED pandas/plotting/_misc.py::pandas.plotting._misc.lag_plot
FAILED pandas/plotting/_misc.py::pandas.plotting._misc.parallel_coordinates
FAILED pandas/plotting/_misc.py::pandas.plotting._misc.radviz
FAILED pandas/plotting/_misc.py::pandas.plotting._misc.scatter_matrix
```
Not many, I'll see if I can create specific issues for some of them.

--- a/ci/code_checks.sh
+++ b/ci/code_checks.sh
@@ -75,6 +75,7 @@ if [[ -z "$CHECK" || "$CHECK" == "doctests" ]]; then
       pandas/core \
       pandas/errors/ \
       pandas/io/ \
+      pandas/plotting/ \
       pandas/tseries/ \
       pandas/util/ \
       pandas/_typing.py \

--- a/pandas/plotting/_core.py
+++ b/pandas/plotting/_core.py
@@ -338,7 +338,7 @@ by ``df.boxplot()`` or indicating the columns to be used:
     >>> np.random.seed(1234)
     >>> df = pd.DataFrame(np.random.randn(10, 4),
     ...                   columns=['Col1', 'Col2', 'Col3', 'Col4'])
-    >>> boxplot = df.boxplot(column=['Col1', 'Col2', 'Col3'])
+    >>> boxplot = df.boxplot(column=['Col1', 'Col2', 'Col3'])  # doctest: +SKIP
 
 Boxplots of variables distributions grouped by the values of a third
 variable can be created using the option ``by``. For instance:
@@ -381,7 +381,7 @@ or changing the fontsize (i.e. ``fontsize=15``):
 .. plot::
     :context: close-figs
 
-    >>> boxplot = df.boxplot(grid=False, rot=45, fontsize=15)
+    >>> boxplot = df.boxplot(grid=False, rot=45, fontsize=15)  # doctest: +SKIP
 
 The parameter ``return_type`` can be used to select the type of element
 returned by `boxplot`.  When ``return_type='axes'`` is selected,
@@ -591,14 +591,14 @@ def boxplot_frame_groupby(
         >>> data = np.random.randn(len(index),4)
         >>> df = pd.DataFrame(data, columns=list('ABCD'), index=index)
         >>> grouped = df.groupby(level='lvl1')
-        >>> grouped.boxplot(rot=45, fontsize=12, figsize=(8,10))
+        >>> grouped.boxplot(rot=45, fontsize=12, figsize=(8,10))  # doctest: +SKIP
 
     The ``subplots=False`` option shows the boxplots in a single figure.
 
     .. plot::
         :context: close-figs
 
-        >>> grouped.boxplot(subplots=False, rot=45, fontsize=12)
+        >>> grouped.boxplot(subplots=False, rot=45, fontsize=12)  # doctest: +SKIP
     """
     plot_backend = _get_plot_backend(backend)
     return plot_backend.boxplot_frame_groupby(
@@ -987,6 +987,7 @@ class PlotAccessor(PandasObject):
 
             >>> s = pd.Series([1, 3, 2])
             >>> s.plot.line()
+            <AxesSubplot:ylabel='Density'>
 
         .. plot::
             :context: close-figs

--- a/pandas/plotting/_matplotlib/groupby.py
+++ b/pandas/plotting/_matplotlib/groupby.py
@@ -49,8 +49,15 @@ def create_iter_data_given_by(
     ...          [3, 4, np.nan, np.nan], [np.nan, np.nan, 5, 6]]
     >>> data = DataFrame(value, columns=mi)
     >>> create_iter_data_given_by(data)
-    {'h1': DataFrame({'a': [1, 3, np.nan], 'b': [3, 4, np.nan]}),
-     'h2': DataFrame({'a': [np.nan, np.nan, 5], 'b': [np.nan, np.nan, 6]})}
+    {'h1':     h1
+         a    b
+    0  1.0  3.0
+    1  3.0  4.0
+    2  NaN  NaN, 'h2':     h2
+         a    b
+    0  NaN  NaN
+    1  NaN  NaN
+    2  5.0  6.0}
     """
 
     # For `hist` plot, before transformation, the values in level 0 are values
@@ -96,10 +103,10 @@ def reconstruct_data_with_by(
     >>> df = DataFrame(d)
     >>> reconstruct_data_with_by(df, by='h', cols=['a', 'b'])
        h1      h2
-       a   b   a   b
-    0  1   3   NaN NaN
-    1  3   4   NaN NaN
-    2  NaN NaN 5   6
+       a     b     a     b
+    0  1.0   3.0   NaN   NaN
+    1  3.0   4.0   NaN   NaN
+    2  NaN   NaN   5.0   6.0
     """
     grouped = data.groupby(by)
 

--- a/pandas/plotting/_misc.py
+++ b/pandas/plotting/_misc.py
@@ -123,6 +123,22 @@ def scatter_matrix(
 
         >>> df = pd.DataFrame(np.random.randn(1000, 4), columns=['A','B','C','D'])
         >>> pd.plotting.scatter_matrix(df, alpha=0.2)
+        array([[<AxesSubplot:xlabel='A', ylabel='A'>,
+            <AxesSubplot:xlabel='B', ylabel='A'>,
+            <AxesSubplot:xlabel='C', ylabel='A'>,
+            <AxesSubplot:xlabel='D', ylabel='A'>],
+           [<AxesSubplot:xlabel='A', ylabel='B'>,
+            <AxesSubplot:xlabel='B', ylabel='B'>,
+            <AxesSubplot:xlabel='C', ylabel='B'>,
+            <AxesSubplot:xlabel='D', ylabel='B'>],
+           [<AxesSubplot:xlabel='A', ylabel='C'>,
+            <AxesSubplot:xlabel='B', ylabel='C'>,
+            <AxesSubplot:xlabel='C', ylabel='C'>,
+            <AxesSubplot:xlabel='D', ylabel='C'>],
+           [<AxesSubplot:xlabel='A', ylabel='D'>,
+            <AxesSubplot:xlabel='B', ylabel='D'>,
+            <AxesSubplot:xlabel='C', ylabel='D'>,
+            <AxesSubplot:xlabel='D', ylabel='D'>]], dtype=object)
     """
     plot_backend = _get_plot_backend("matplotlib")
     return plot_backend.scatter_matrix(
@@ -208,6 +224,7 @@ def radviz(frame, class_column, ax=None, color=None, colormap=None, **kwds):
         ...     }
         ... )
         >>> pd.plotting.radviz(df, 'Category')
+        <AxesSubplot:xlabel='y(t)', ylabel='y(t + 1)'>
     """
     plot_backend = _get_plot_backend("matplotlib")
     return plot_backend.radviz(
@@ -266,6 +283,7 @@ def andrews_curves(
         ...     'pandas/master/pandas/tests/io/data/csv/iris.csv'
         ... )
         >>> pd.plotting.andrews_curves(df, 'Name')
+        <AxesSubplot:title={'center':'width'}>
     """
     plot_backend = _get_plot_backend("matplotlib")
     return plot_backend.andrews_curves(
@@ -325,6 +343,7 @@ def bootstrap_plot(series, fig=None, size=50, samples=500, **kwds):
 
         >>> s = pd.Series(np.random.uniform(size=100))
         >>> pd.plotting.bootstrap_plot(s)
+        <Figure size 640x480 with 6 Axes>
     """
     plot_backend = _get_plot_backend("matplotlib")
     return plot_backend.bootstrap_plot(
@@ -392,6 +411,7 @@ def parallel_coordinates(
         >>> pd.plotting.parallel_coordinates(
         ...     df, 'Name', color=('#556270', '#4ECDC4', '#C7F464')
         ... )
+        <AxesSubplot:xlabel='y(t)', ylabel='y(t + 1)'>
     """
     plot_backend = _get_plot_backend("matplotlib")
     return plot_backend.parallel_coordinates(
@@ -440,6 +460,7 @@ def lag_plot(series, lag=1, ax=None, **kwds):
         >>> x = np.cumsum(np.random.normal(loc=1, scale=5, size=50))
         >>> s = pd.Series(x)
         >>> s.plot()
+        <AxesSubplot:xlabel='Midrange'>
 
     A lag plot with ``lag=1`` returns
 
@@ -447,6 +468,7 @@ def lag_plot(series, lag=1, ax=None, **kwds):
         :context: close-figs
 
         >>> pd.plotting.lag_plot(s, lag=1)
+        <AxesSubplot:xlabel='y(t)', ylabel='y(t + 1)'>
     """
     plot_backend = _get_plot_backend("matplotlib")
     return plot_backend.lag_plot(series=series, lag=lag, ax=ax, **kwds)
@@ -480,6 +502,7 @@ def autocorrelation_plot(series, ax=None, **kwargs):
         >>> spacing = np.linspace(-9 * np.pi, 9 * np.pi, num=1000)
         >>> s = pd.Series(0.7 * np.random.rand(1000) + 0.3 * np.sin(spacing))
         >>> pd.plotting.autocorrelation_plot(s)
+        <AxesSubplot:title={'center':'width'}, xlabel='Lag', ylabel='Autocorrelation'>
     """
     plot_backend = _get_plot_backend("matplotlib")
     return plot_backend.autocorrelation_plot(series=series, ax=ax, **kwargs)

